io/plugins/eu.esdihumboldt.hale.io.gml/src/eu/esdihumboldt/hale/io/gml/writer/internal/StreamGmlWriter.java

```diff
@@ -1602,6 +1604,14 @@ private void writeProperties(Group parent, Collection<? extends ChildDefinition<
 
 			if (child.asProperty() != null) {
 				PropertyDefinition propDef = child.asProperty();
+
+				boolean isInspireType = GmlWriterUtil.isInspireType(propDef);
```

Comment:
You can't determine if the parent of the property is an INSPIRE type by checking the property.

You have the parent group (`parent`) available in this method. What you could do is check if it is an instance of the `Instance` class, and in that case extract the parent `TypeDefinition`. That you could check if it is an INSPIRE type.